### DISCUSSION TITLE
Option to set JDK and JAVA_HOME through scala-cli

### DISCUSSION THREAD

**Discussion by yadavan88** [VirtusLab/scala-cli#2764] [Ideas]
Hi, 
I want to check if this is possible. 
I use coursier a lot to setup my environment especially switching between jdks (cs java --jvm 11 --setup). I also promote cs whenever I get chance (either in meetups or to colleagues and friends).

Since scala-cli already uses cs for its own jdks, is there any way to allow to set the JDK version in the system as well? 

The reason is that many ppl have scala-cli installed now and if we can utilize some power flags to redirect the cs command for java setup as well, it makes much easier to manage JVMs. 

Not sure if this is a stupid idea, but would like to know your opinion.

**Comment by Gedochao** [comment]
We have seen suggestions to expose Coursier features in Scala CLI (https://github.com/VirtusLab/scala-cli/issues/2047, https://github.com/VirtusLab/scala-cli/issues/2172).
There are no immediate plans to do it, but I think it's not unlikely we'll end up doing it at some point.

As for managing the JDKs straight from Scala CLI, I'd say if we were to support it, it should definitely be done via `coursier`. 
Feel free to create a feature request issue and see if it gathers interest. 
That'd be another reason to expose `cs` in some way or form in the future.
You can also bump one of the existing ticket along with describing your usecase.
